src/sentry/sentry_metrics/consumers/indexer/batch.py

```diff
@@ -144,6 +145,17 @@ def _extract_messages(self) -> None:
                     exc_info=True,
                 )
                 continue
+            if (
+                datetime.fromtimestamp(parsed_payload["timestamp"], tz=timezone.utc)
+                - datetime.now(tz=timezone.utc)
```

Comment:
```suggestion
                time.time() - parsed_payload("timestamp"]
```

- no need to convert to datetimes
- now + parsed timestamp flipped around the wrong way